deep_translator/google.py

```diff
@@ -120,3 +123,53 @@ def translate_batch(self, batch: List[str], **kwargs) -> List[str]:
         @return: list of translations
         """
         return self._translate_batch(batch, **kwargs)
+
+    @lru_cache(maxsize=None)
+    async def _async_translate(
+        self, text: str, session: aiohttp.ClientSession, **kwargs
+    ):
+        if is_input_valid(text):
```

Comment:
Since most of the code in this function is the same as the non-async translate, maybe you can find a way to make some parts reusable

Reply:
@nidhaloff I'll  try to keep all the things in mind and fix it. Thanks for you detailed explanation.